app/controllers/users_controller.rb

```diff
@@ -1,3 +1,27 @@
 class UsersController < ApplicationController
-  def new; end
+  def new
+    @user = User.new
+  end
+
+  def show
+    @user = User.find_by id: params[:id]
+    redirect_to root_path if @user.nil?
```

Comment:
Nên có message để thân thiện ng dùng nhé. Chìa tay cũng cần lý do mà
```suggestion
    return if @user
    flash[.....
    redirect_to root_path 
```